### PR TITLE
chore: improve home view screen memory consumption

### PR DESCRIPTION
**Pull request by MounirDhahri**
[future-friday]

### Description

This PR updates the default rendering values in the home view rails in order to improve the performance and memory consumption.

By default, `initialNumToRender` is 10 and `windowSize` is 21. That's fine for vertical lists where more content is visible. In the case of our horizontal lists, we barely show 2 items, and we don't need to render the 10th at initial render.

| Before | After |
|--------|--------|
| Screenshot 2024-09-13 at 18 02 41 | Screenshot 2024-09-13 at 18 00 12 | 

**What comes next?**

- Maybe we can move the constants to a more shared folder and use them in other areas of the app
- We can probably be even more aggressive with some rails like shows, fairs and collections which are heavy and be less strict with artwork rails.

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- improve home view screen memory consumption - mounir

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (improve home view screen memory consumption - mounir)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c37b56afd0077714f1e9bd07ecc88df2215b924c
</p>

**Review by damassi — DISMISSED** [review]
👌 👌

File: src/app/Components/ArtworkRail/ArtworkRail.tsx [hunk 1]
```diff
@@ -92,7 +96,8 @@ export const ArtworkRail: React.FC<ArtworkRailProps> = ({
       }
       showsHorizontalScrollIndicator={false}
       data={artworksSlice}
-      initialNumToRender={isTablet() ? 10 : 5}
+      initialNumToRender={isTablet() ? 10 : HORIZONTAL_FLATLIST_INTIAL_NUMBER_TO_RENDER_ARTWORKS}
```
Comment (nickskalkin) [inline src/app/Components/ArtworkRail/ArtworkRail.tsx:99]: **suggestion / non-blocking**: while at it - maybe extract 10 to a constant as well?

**Review by nickskalkin — APPROVED** [review]
🙌

File: src/app/Components/ArtworkRail/ArtworkRail.tsx (hunk 1 above)
Comment (MounirDhahri) [reply src/app/Components/ArtworkRail/ArtworkRail.tsx:99]: yes, good idea - I will do that

**Comment by anandaroop** [comment]
Very nice, thanks.

Out of curiosity, this profiling is done in XCode? A quick KS or show-and-tell on that would be great!

**Pull request merged**